### PR TITLE
capnproto/0.7.0

### PR DESCRIPTION
**Pull request by blackliner**
Specify library name and version:  **capnproto/0.7.0**

Add support for the older version 0.7.0

ARC-119

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by blackliner** [comment]
@prince-chrismc any advice on that symlink issue? Is this related to the recipe/build or something that might have changed since the introduction of this capnproto recipe?

**Comment by prince-chrismc** [comment]
No expert here... I think you are missing this https://github.com/capnproto/capnproto/commit/fdbf035619ab2f9e25173bb7361e7e19a52e0fa1

0.7.0 https://github.com/capnproto/capnproto/blame/3f44c6db0f0f6c0cab0633f15f15d0a2acd01d19/c%2B%2B/src/capnp/CMakeLists.txt#L185

0.8.0

**Comment by SpaceIm** [comment]
What about all the patches of 0.8.0? There is little chance that you could provide 0.7.0 without similar patches.

**Comment by blackliner** [comment]
> What about all the patches of 0.8.0? There is little chance that you could provide 0.7.0 without similar patches.

It builds fine on my end with clang and gcc, so I wanted to know what the issue might be.

**Comment by blackliner** [comment]
> No expert here... I think you are missing this [capnproto/capnproto@fdbf035](https://github.com/capnproto/capnproto/commit/fdbf035619ab2f9e25173bb7361e7e19a52e0fa1)
> 
> 0.7.0 https://github.com/capnproto/capnproto/blame/3f44c6db0f0f6c0cab0633f15f15d0a2acd01d19/c%2B%2B/src/capnp/CMakeLists.txt#L185
> 
> 0.8.0
> https://github.com/capnproto/capnproto/blame/57a4ca5af5a7f55b768a9d9d6655250bffb1257f/c%2B%2B/src/capnp/CMakeLists.txt#L189
> 
> The commit above changed the extension for windows.... which is what failed when looking for the file missing an extension ... make sense but I could be wrong

Thanks @prince-chrismc , that was it 👍

**Comment by prince-chrismc** [comment]
> > What about all the patches of 0.8.0? There is little chance that you could provide 0.7.0 without similar patches.
> 
> It builds fine on my end with clang and gcc, so I wanted to know if that is required at all.

Sometimes build is not enough, we do not test everything so there's a little judgement we need to make... I am not familiar with this library, but naively I think make 1, 2, and 4 are important enough they are required for CCI

**Comment by blackliner** [comment]
@prince-chrismc I added patch1 and 4, but 2 is not possible, as the compat stuff completely changed, and actually looks similar to how the result of the patch would look like

**Comment by prince-chrismc** [comment]
Seems like KJ-gzip is just a c file not a library 🤔

**Comment by blackliner** [comment]
@prince-chrismc I think its dead, Jim

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/capnproto/all/conanfile.py [hunk 1]
```diff
@@ -67,9 +70,9 @@ def configure(self):
             raise ConanInvalidConfiguration("Cap'n Proto doesn't support shared libraries for Visual Studio")
 
     def requirements(self):
-        if self.options.with_openssl:
+        if self.options.get_safe("with_openssl"):
```
Comment (SpaceIm) [inline recipes/capnproto/all/conanfile.py:73]:
```suggestion
        if self.options.with_openssl:
```

File: recipes/capnproto/all/conanfile.py [hunk 2]
```diff
@@ -97,8 +100,8 @@ def _configure_autotools(self):
         args = [
             "--enable-shared" if self.options.shared else "--disable-shared",
             "--disable-static" if self.options.shared else "--enable-static",
-            "--with-openssl" if self.options.with_openssl else "--without-openssl",
-            "--with-zlib" if self.options.with_zlib else "--without-zlib",
+            "--with-openssl" if self.options.get_safe("with_openssl") else "--without-openssl",
+            "--with-zlib" if self.options.get_safe("with_zlib") else "--without-zlib",
             "--enable-reflection"
         ]
```
Comment (SpaceIm) [inline recipes/capnproto/all/conanfile.py:106]:
```suggestion
            "--with-openssl" if self.options.with_openssl else "--without-openssl",
            "--enable-reflection"
        ]
        if tools.Version(self.version) >= "0.8.0":
            args.append("--with-zlib" if self.options.with_zlib else "--without-zlib")
```

File: recipes/capnproto/all/conanfile.py [hunk 3]
```diff
@@ -157,9 +160,9 @@ def package_info(self):
             {"name": "kj-http", "requires": ["kj", "kj-async"]},
             {"name": "kj-test", "requires": ["kj"]},
         ]
-        if self.options.with_zlib:
+        if self.options.get_safe("with_zlib"):
             components.append({"name": "kj-gzip", "requires": ["kj", "kj-async", "zlib::zlib"]})
-        if self.options.with_openssl:
+        if self.options.get_safe("with_openssl"):
```
Comment (SpaceIm) [inline recipes/capnproto/all/conanfile.py:165]:
```suggestion
        if self.options.with_openssl:
```

**Comment by prince-chrismc** [comment]
Please do not force push! Github makes us restart reviewing 😭

**Review by prince-chrismc — COMMENTED** [review]
I assume TLS is important for this project, can you try Space's comments?

File: recipes/capnproto/all/conanfile.py [hunk 4]
```diff
@@ -87,7 +90,7 @@ def _configure_cmake(self):
         self._cmake.definitions["BUILD_TESTING"] = False
         self._cmake.definitions["EXTERNAL_CAPNP"] = False
         self._cmake.definitions["CAPNP_LITE"] = False
-        self._cmake.definitions["WITH_OPENSSL"] = self.options.with_openssl
+        self._cmake.definitions["WITH_OPENSSL"] = self.options.get_safe("with_openssl")
```
Comment (SpaceIm) [inline recipes/capnproto/all/conanfile.py:93]:
```suggestion
        self._cmake.definitions["WITH_OPENSSL"] = self.options.with_openssl
```

**Comment by blackliner** [comment]
6fba328ef0a51f07737c42491cd37fa127e4502e was the commit where I only disabled zlib, but that was not enough, CI failed with a similar message as with zlib for TLS

**Comment by prince-chrismc** [comment]
> [6fba328](https://github.com/conan-io/conan-center-index/commit/6fba328ef0a51f07737c42491cd37fa127e4502e) was the commit where I only disabled zlib, but that was not enough, CI failed with a similar message as with zlib for TLS

Well https://github.com/capnproto/capnproto/commit/7da6abf233fdf74a4ad4db56b6e98778b162dd7d

Was added 2 years after this old release.... I don't know why linux and mac pass....

**Comment by SpaceIm** [comment]
Sure it fails, it requires similar patch than 0.8.0

**Comment by blackliner** [comment]
> Sure it fails, it requires similar patch than 0.8.0: https://github.com/conan-io/conan-center-index/blob/3df626e745868dfdba5248cf58a3f8190dd157d7/recipes/capnproto/all/patches/0003-kj-tls-windows.patch

I dont need the TLS for 0.7.0, do yo insist on patching it for windows?

**Comment by prince-chrismc** [comment]
Nope! Just can revert and either 
- add a fix me comment
- or throw when on and windows this way linux gets it

**Comment by SpaceIm** [comment]
I would prefer to use the same patch than 0.8.0, it's easy and avoids to much complexity and weird logic in conanfile.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Pull request merged**